### PR TITLE
Iterators and Generators 번역 #53

### PR DESCRIPTION
**Pull request by ye-geeee**

### CODE REVIEW

File: iterators-and-generators.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# 반복자와 생성자 (Iterators and Generators)
+
+## 반복 가능 (Iterables)
+
+[`Symbol.iterator`](https://www.typescriptlang.org/docs/handbook/symbols.html#symboliterator) 속성에 대한 구현이 있는 객체는 반복 가능한 것으로 간주합니다. 
+`Array`, `Map`, `Set`, `String`, `Int32Array`, `Uint32Array` 등과 같은 일부 내장된(built-in) 타입에는 이미 `Symbol.iterator` 속성이 구현되어 있습니다. 
+객체의 `Symbol.iterator` 함수는 반복할 값 목록을 반환합니다.
+
+### for .. of 문
```
Comment (bumkeyy) [inline iterators-and-generators.md:9]:
```suggestion
### `for .. of` 문
```

File: iterators-and-generators.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+for..of는 반복 가능한 객체를 반복하여 객체의 `Symbol.iterator` 속성을 호출합니다.
```
Comment (Bumkeyy) [inline iterators-and-generators.md]:
```suggestion
`for..of`는 반복 가능한 객체를 반복하여 객체의 `Symbol.iterator` 속성을 호출합니다. 
```

Comment (bumkeyy) [inline iterators-and-generators.md:11]:
```suggestion
`for..of`는 반복 가능한 객체를 반복하여 객체의 `Symbol.iterator` 속성을 호출합니다. 
```

File: iterators-and-generators.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+## 반복 가능 (Iterables)
+
+[`Symbol.iterator`](https://www.typescriptlang.org/docs/handbook/symbols.html#symboliterator) 속성에 대한 구현이 있는 객체는 반복 가능한 것으로 간주합니다. 
+`Array`, `Map`, `Set`, `String`, `Int32Array`, `Uint32Array` 등과 같은 일부 내장된(built-in) 타입에는 이미 `Symbol.iterator` 속성이 구현되어 있습니다. 
+객체의 `Symbol.iterator` 함수는 반복할 값 목록을 반환합니다.
+
+### for .. of 문
+
+for..of는 반복 가능한 객체를 반복하여 객체의 `Symbol.iterator` 속성을 호출합니다. 
+다음은 배열의 간단한 for..of 루프입니다.
```
Comment (bumkeyy) [inline iterators-and-generators.md:12]:
```suggestion
다음은 배열의 간단한 `for..of` 루프입니다.
```

File: iterators-and-generators.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+```ts
+let someArray = [1, "string", false];
+
+for (let entry of someArray){
+ console.log(entry); // 1, "string", false
+}
+```
+
+#### _for..of vs. for..in 문_
```
Comment (bumkeyy) [inline iterators-and-generators.md:22]:
```suggestion
#### _`for..of` vs. `for..in` 문_
```

**Comment by bumkeyy** [comment]
고생하셨습니다 👍

File: iterators-and-generators.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+for (let pet of pets){
+ console.log(pet); // "Cat", "Dog", "Hamster"
+}
+```
+
+_코드 생성_
+_ES5 및 ES3 타게팅_
+
+
+ES5 또는 ES3 호환 엔진을 대상으로하는 경우, 반복자는 배열 유형의 값에만 허용됩니다.
```
Comment (guyeol) [inline iterators-and-generators.md:63]:
```suggestion
ES5 또는 ES3 호환 엔진을 대상으로하는 경우, 반복자는 배열 타입의 값에만 허용됩니다. 
```

File: iterators-and-generators.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+ console.log(i); // "0", "1", "2"
+}
+
+for (let i of list){
+ console.log(i); // "4", "5", "6"
+}
+```
+
+또 다른 차이점은 `for..in`은 모든 객체에서 작동한다는 것입니다. 
+이 객체의 속성을 검사하는 방법으로 사용됩니다.
```
Comment (guyeol) [inline iterators-and-generators.md:42]:
```suggestion
이 객체의 프로터피들을 검사하는 방법으로 사용됩니다. 
```

**Comment by yeonjuan** [comment]
@ye-geeee  고생하셨습니다!. 👍 

fixes #53 

@all-contributors please add @ye-geeee for translation

**Pull request merged**